test/lib/pgBackRestTest/Info/InfoIniUnitTest.pm

```diff
@@ -0,0 +1,467 @@
+# run
+####################################################################################################################################
+sub run
+{
+    my $self = shift;
+
+    # Test ini file
+    my $strTestFile = $self->testPath() . '/test.ini';
+    my $strTestFileCopy = "${strTestFile}.copy";
+
+    # Test keys, values
+    my $strSection = 'test-section';
+    my $strKey = 'test-key';
+    my $strSubKey = 'test-subkey';
+    my $strValue = 'test-value';
+
+    ################################################################################################################################
+    if ($self->begin("iniRender()"))
+    {
+        #---------------------------------------------------------------------------------------------------------------------------
+        my $oIni = {section1 => {key => 'value'}, section2 => {key => ['value1', 'value2']}};
+
+        $self->testResult(
+            sub {iniRender($oIni, true)}, "[section1]\nkey=value\n\n[section2]\nkey=value1\nkey=value2\n",
+            'relaxed formatting');
+        $self->testResult(
+            sub {iniRender($oIni, false)}, "[section1]\nkey=\"value\"\n\n[section2]\nkey=[\"value1\",\"value2\"]\n",
+            'strict formatting');
+    }
+
+    ################################################################################################################################
+    if ($self->begin("iniParse()"))
+    {
+        #---------------------------------------------------------------------------------------------------------------------------
+        my $strIni = "[section]\nkey";
+
+        $self->testException(sub {iniParse($strIni)}, ERROR_CONFIG, "unable to find '=' in 'key'");
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $strIni = "[section]\n# comment\n\nkey=\"value\"";
+
+        $self->testResult(sub {iniParse($strIni)}, '{section => {key => value}}', 'ignore blank lines and comments');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $strIni = "[section]\nkey=value";
+
+        $self->testResult(sub {iniParse($strIni, {bRelaxed => true})}, '{section => {key => value}}', 'relaxed value read');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $strIni = "[section]\nkey=value1\nkey=value2\nkey=value3";
+
+        $self->testResult(
+            sub {iniParse($strIni, {bRelaxed => true})}, '{section => {key => (value1, value2, value3)}}', 'relaxed array read');
+    }
+
+    ################################################################################################################################
+    if ($self->begin("Ini->new()"))
```

Comment:
Good question - you actually get a checksum error.  I have added a new error message.